src/Queue.php

```diff
@@ -155,14 +152,7 @@ public function get(array $query, $runningResetDuration, $waitDurationInMillis =
             $pollDurationInMillis = 0;
         }
 
-        //reset stuck messages
-        $this->collection->update(
-            ['running' => true, 'resetTimestamp' => ['$lte' => new \MongoDate()]],
-            ['$set' => ['running' => false]],
-            ['multiple' => true]
```

Comment:
The first update does the reset for all needed docs on the collection, the findandmodify does only one, and only if a boolean condition for running true were added to the query, which would take an OR and break the index.